src/gsi/setupfed.f90

```diff
@@ -0,0 +1,1152 @@
+  call init_vars_
+!  qscalar=zero 
+ 
+ !============================================================================================
+ ! 1) Calculate the graupel-mass and graupel-volume based flash extent density
+ ! (FED) on model space, added by R. Kong, 07/05/2018
+ !============================================================================================
+  ges_qg_mask=ges_qg
+  where(ges_qg>0.0005)  !Count the volume where qg > 0.5/kg
+     ges_qg_mask=1.0
+  elsewhere
+     ges_qg_mask=0.0
+  endwhere
+
+  ! Operator start here
+  ! set ngx and ngy =2, so the integrated domain is 15kmx15km
+  ngx = 2
+  ngy = 2
+  dx_m = 3000.
+  dy_m = 3000.
+  print*,'Operator start here!,ngx=',ngx,'ngy=',ngy
+  rp=zero
+
+  print*, 'mype = ', mype
+  print*, 'nfldsig = ', nfldsig
+  print*, 'nsig = ', nsig
+  print*, 'lon2 = ', lon2
+  print*, 'lat2 = ', lat2
+
+! compute graupel mass, in kg per 15 km x 15 km column
+  do jj=1,nfldsig
+     do k=1,nsig
+        do i=1,lon2
+           do j=1,lat2   !How to handle MPI????
+              do igx=1,2*ngx+1 !horizontal integration of qg around point to calculate FED
+                do jgy=1,2*ngy+1
+                  itmp = i-ngx+igx-1
+                  jtmp = j-ngy+jgy-1
+                  itmp = min(max(1,itmp),lon2)
+                  jtmp = min(max(1,jtmp),lat2)
+                    rp(j,i,jj)=rp(j,i,jj) + ges_qg(jtmp,itmp,k,jj)* &
+                               dx_m*dy_m*(ges_prsi(jtmp,itmp,k,jj)-ges_prsi(jtmp,itmp,k+1,jj))*&
+                               tpwcon * r10
+                end do   !igx
+              end do  !jgy
+           end do !j
+        end do !i
+     end do !k
+  end do !jj
+
+! compute FED, in flashes/min
+  do jj=1,nfldsig
+     do i=1,lon2
+        do j=1,lat2
+           if (fed_obs_ob_shape .eq. 1) then
+              rp(j,i,jj) = CM * glmcoeff * rp(j,i,jj)
+           else if (fed_obs_ob_shape .eq. 2) then
+              rp(j,i,jj) = a_coeff + b_coeff &
+                           * tanh(c_coeff * (rp(j,i,jj) - d_coeff))
+           else
+              write(6,*) ' unknown fed_obs_ob_shape: ', fed_obs_ob_shape
+              write(6,*) ' aborting setupfed'
+              call stop2(999)
+           end if
+           if (rp(j,i,jj) .gt. fed_highbnd) rp(j,i,jj) = fed_highbnd
+        end do !j
+     end do !i
+  end do !jj
+
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+     write(6,*) 'fed_obs_ob_shape=',fed_obs_ob_shape
+     if (fed_obs_ob_shape .eq. 2) then
+       write(6,*) 'a_coeff=',a_coeff
+       write(6,*) 'b_coeff=',b_coeff
+       write(6,*) 'c_coeff=',c_coeff
+       write(6,*) 'd_coeff=',d_coeff
+     end if
+     write(6,*) 'fed_highbnd=',fed_highbnd
+     write(6,*) 'maxval(ges_qg)=',maxval(ges_qg),'pe=',mype
+!     write(6,*) 'maxval(geop_hgtl)=',maxval(geop_hgtl(:,:,:,it))
+     write(6,*) 'maxval(ges_tsen)=',maxval(ges_tsen(:,:,:,it))
```

Comment:
@daviddowellNOAA @hu5970 Please delete or commented out the below two lines: 
    write(6,*) 'maxval(ges_tsen)=',maxval(ges_tsen(:,:,:,it))
    write(6,*) 'ges_prsi',ges_prsi(100,100,1,1),ges_prsi(100,100,nsig,1)
The ges_tsen will crash GSI in the debug mode. Thanks.